packages/vite/src/node/optimizer/esbuildDepPlugin.ts

```diff
@@ -268,19 +271,36 @@ export function esbuildCjsExternalPlugin(externals: string[]): Plugin {
         `^${text.replace(/[-/\\^$*+?.()|[\]{}]/g, '\\$&')}$`
       const filter = new RegExp(externals.map(escape).join('|'))
 
-      build.onResolve({ filter: /.*/, namespace: 'external' }, (args) => ({
-        path: args.path,
-        external: true
-      }))
+      build.onResolve({ filter: new RegExp(`^${nonFacadePrefix}`) }, (args) => {
+        return {
+          path: args.path.slice(nonFacadePrefix.length),
+          external: true
+        }
+      })
+
+      build.onResolve({ filter }, (args) => {
+        if (args.kind === 'require-call') {
+          return {
+            path: args.path,
+            namespace: cjsExternalFacadeNamespace
+          }
+        }
 
-      build.onResolve({ filter }, (args) => ({
-        path: args.path,
-        namespace: 'external'
-      }))
+        return {
+          path: args.path,
+          external: true
+        }
+      })
 
-      build.onLoad({ filter: /.*/, namespace: 'external' }, (args) => ({
-        contents: `export * from ${JSON.stringify(args.path)}`
-      }))
+      build.onLoad(
+        { filter: /.*/, namespace: cjsExternalFacadeNamespace },
+        (args) => ({
+          contents:
+            `import * as m from ${JSON.stringify(
+              nonFacadePrefix + args.path
+            )};` + `module.exports = m;`
```

Comment:
I changed to `module.exports` from `export *` + `export { default }`.

This will reduce one conversion.
```js
// --- transform from
// foo
module.exports = 'foo'

// bar
const foo = require('foo')

// --- transform to (previously)
// foo:facade
export * from 'foo'
export { default } from 'foo'

// bar
const foo = require('foo')

// --- transform to
// foo:facade
import * as m from 'foo'
module.exports = m

// bar
const foo = require('foo')
```

Reply:
I guess it's worth trying this out if it works 👍

Reply:
@bluwy @sapphi-red have been using a similar patch in a medium sized project for a while now without issues, only difference is that I am using the below instead.
```
import m from ${JSON.stringify(
  nonFacadePrefix + args.path
)};` + `module.exports = m;`
```

Reply:
@jiby-aurum Your case would be covered by #11057.

Reply:
@sapphi-red wow didn't see that one, thanks !!